pkg/networkservice/common/monitor/test_monitor.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitor
+
+import (
+	"context"
+	"runtime"
+	"time"
+
+	"github.com/networkservicemesh/sdk/pkg/tools/serialize"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+)
+
+// TestMonitorClient - implementation of test monitor client.
+type TestMonitorClient struct {
+	Events       []*networkservice.ConnectionEvent
+	eventChannel chan *networkservice.ConnectionEvent
+	ctx          context.Context
+	Cancel       context.CancelFunc
+	grpc.ServerStream
+	executor  serialize.Executor
+	finalized chan struct{}
+}
+
+// NewTestMonitorClient - construct a new client.
+func NewTestMonitorClient() *TestMonitorClient {
```

Comment:
Do you intend TestMonitorClient to be a useful tool for testing things outside of the monitor package?

Reply:
@lobkovilya wanted to use it for some of healing tests AS I remember. I could make it private for now. And change back if it will be required.

Reply:
Lets do that :)

Reply:
No, I suggested to put auxiliary classes (fixtures, mock, dummy and etc) to `pkg/test` to have ability use them from another unit tests. Package name should reflect the fact that this is testing util - `monitortestutil` or smth like that. 
I think it would be nice to keep `*_test.go` files without any auxiliary stuff, just unit tests.

Reply:
I'm fine with doing that *when* we come across a case for which we want to reuse... but lets work with actual rather than imagined cases of that sort.  Keep things as simple as possible :)

Reply:
Yes but in the future someone might doesn't notice that `TestMonitorClient` implementation and write own one. So I think we should keep all auxiliary testing stuff in one place.